README


This archive contains the distribution Alien-MuPDF,
version 0.008:

  Alien package for the MuPDF PDF rendering library

This software is copyright (c) 2013 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.008.



Changes
0.008 2017-05-14 15:58:04-0500
  - No changes since 0.007_02.

0.007_02 2017-05-09 08:35:58-0500
  - Switch to alienfile and Alien::Build.

    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/31>.

0.007_01 2017-04-23 19:10:32-0500
  - Compile MuPDF so that it can be linked into a shared library (e.g., with XS)
    by setting `XCFLAGS=-fPIC`.

    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/24>.

  - Sets the Inline parameter `MYEXTLIB` to the path for the mupdf archive
    files so that it can be found via Inline::C.

  - Remove the FreeBSD patch to `fontdump.c` that is no longer needed with
    mupdf-1.11.

    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/32>,
    RT#121218 <https://rt.cpan.org/Ticket/Display.html?id=121218>.
    Thanks to SREZIC++ for the bug report.

0.007 2016-11-23 02:10:08-0600
  - Fix the t/inline.t version number test so that it works with
    the MuPDF v1.10 release.

    Fixes RT#118882 <https://rt.cpan.org/Ticket/Display.html?id=118882>,
    <https://github.com/project-renard/p5-Alien-MuPDF/issues/22>.
    Thanks to SREZIC++ for the bug report.

0.006 2016-08-25 05:52:48-0500
  - Change syntax in test so that it runs on Perl v5.8
    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/20>.

0.005_03 2016-08-21 13:40:09-0500
  - Use File::Which to get path to mutool binary in t/use.t. This allows for
    checking extensions (.exe) on Windows systems.
    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/18>.

0.005_02 2016-08-06 18:43:25-0500
  - Apply patch that removes `HAVE_INCBIN` define on FreeBSD. This allows for
    building on FreeBSD's C compiler.
    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/16>.

0.005_01 2016-07-25 18:40:54-0500
  - Fix the path to `mutool` during build-time.
    Fixes <https://github.com/project-renard/p5-Alien-MuPDF/issues/12>.
  - Build: Use Alien::gmake to build on more *nix systems.
    Fixes <https://rt.cpan.org/Ticket/Display.html?id=110287>,
    <https://github.com/project-renard/p5-Alien-MuPDF/issues/11>.
    SREZIC++ for the bug report.

0.005 2016-05-29 06:23:45-0500
  - Add `mutool_path` method to get the path the `mutool` binary
    <https://github.com/project-renard/p5-Alien-MuPDF/issues/6>.
  - Fix the Inline::C `INC` and `AUTO_INCLUDE` parameters
    <https://github.com/project-renard/p5-Alien-MuPDF/issues/7>.

0.004 2016-04-14 20:13:44-0500
  - Actual release. Forgot to pull upstream changes. Eek!
  - Changes are as listed in v0.003.

0.003 2016-04-14 20:03:25-0500
  - Improved handling for intermediate releases (e.g., version 1.7a).
  - Remove strict build requirement for OpenGL viewer.
  - Add wrapper for adding the `mutool` binary to the path.
  - Added a minimum version version for Alien::Base `configure_requires`.
    Thanks to Graham Ollis (PLICEASE++)!

0.002 2015-01-24 14:41:49-0600
  - add ExtUtils::Depends support

0.001 2013-07-15 06:50:14-0500
  - first version
  - packages MuPDF v1.2


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Alien package for the MuPDF PDF rendering library",
  "AUTHOR" => "Zakariyya Mughal <zmughal\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "Alien::gmake" => "0.14",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "DISTNAME" => "Alien-MuPDF",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Alien::MuPDF",
  "PREREQ_PM" => {
    "Alien::Base" => "0.038",
    "File::Spec" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Cwd" => 0,
    "File::Basename" => 0,
    "File::Which" => "1.21",
    "Module::Load" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.008",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Alien::Base" => "0.038",
  "Alien::Build" => "0.32",
  "Alien::Build::MM" => "0.32",
  "Alien::gmake" => "0.14",
  "Capture::Tiny" => 0,
  "Cwd" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "File::Basename" => 0,
  "File::Spec" => 0,
  "File::Which" => "1.21",
  "Module::Load" => 0,
  "Test::More" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
sub MY::postamble {
  $abmm->mm_postamble;
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild



lib/Alien/MuPDF.pm
package Alien::MuPDF;
$Alien::MuPDF::VERSION = '0.008';
use strict;
use warnings;

use parent qw(Alien::Base);
use File::Spec;

sub mutool_path {
	my ($self) = @_;
	File::Spec->catfile( File::Spec->rel2abs($self->dist_dir) , 'bin', 'mutool' );
}

sub inline_auto_include {
	return  [ 'mupdf/fitz.h' ];
}

sub cflags {
	my ($self) = @_;
	my $top_include = File::Spec->catfile( File::Spec->rel2abs($self->dist_dir), qw(include) );
	# We do not include $self->SUPER::cflags() because that adds too many
	# header files to the path. In particular, it adds -Imupdf/fitz, which
	# leads to "mupdf/fitz/math.h" being included when trying to include
	# the C standard "math.h" header.
	return "-I$top_include";
}

sub libs {
	# third party
	"-lcrypto";
}

sub Inline {
	my ($self, $lang) = @_;

	if('C') {
		my $params = Alien::Base::Inline(@_);
		$params->{MYEXTLIB} .= ' ' .
			join( " ",
				map { File::Spec->catfile(
					File::Spec->rel2abs(Alien::MuPDF->dist_dir),
					'lib',  $_ ) }
				qw(libmupdf.a libmupdfthird.a)
			);

		return $params;
	}
}

1;

=pod

=encoding UTF-8

=head1 NAME

Alien::MuPDF - Alien package for the MuPDF PDF rendering library

=head1 VERSION

version 0.008

=head1 METHODS

=head2 mutool_path

Returns a C<Str> which contains the absolute path
to the C<mutool> binary.

=head1 Inline support

This module supports L<Inline's with functionality|Inline/"Playing 'with' Others">.

=head1 SEE ALSO

L<MuPDF|http://mupdf.com/>

=head1 AUTHOR

Zakariyya Mughal <zmughal@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut

__END__
# ABSTRACT: Alien package for the MuPDF PDF rendering library



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::Alien v0.023.
use strict;
use warnings;

use Module::Build 0.28;
use lib qw{inc}; use Alien::Base::ModuleBuild;

my %module_build_args = (
  "alien_bin_requires" => {
    "Alien::gmake" => "0.14"
  },
  "alien_build_commands" => [
    "%{gmake} HAVE_GLFW=no XCFLAGS=-fPIC"
  ],
  "alien_inline_auto_include" => [],
  "alien_install_commands" => [
    "%{gmake} HAVE_GLFW=no XCFLAGS=-fPIC prefix=%s install"
  ],
  "alien_msys" => 1,
  "alien_name" => "MuPDF",
  "alien_repository" => {
    "host" => "mupdf.com",
    "location" => "/downloads/",
    "pattern" => "^mupdf-([\\w\\.]+)-source\\.tar\\.gz\$",
    "protocol" => "http"
  },
  "configure_requires" => {
    "Alien::Base::ModuleBuild" => "0.020",
    "File::ShareDir" => "1.03",
    "Path::Class" => "0.013",
    "perl" => "5.006"
  },
  "dist_abstract" => "Alien package for the MuPDF PDF rendering library",
  "dist_author" => [
    "Zakariyya Mughal <zmughal\@cpan.org>"
  ],
  "dist_name" => "Alien-MuPDF",
  "dist_version" => "0.007_01",
  "license" => "perl",
  "module_name" => "Alien::MuPDF",
  "recommends" => {
    "Inline" => "0.67",
    "Inline::C" => "0.62"
  },
  "recursive_test_files" => 1,
  "requires" => {
    "Alien::Base" => "0.020",
    "File::ShareDir" => "1.03",
    "File::Spec" => 0,
    "Path::Class" => "0.013",
    "parent" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [
    "bin/mutool"
  ],
  "test_requires" => {
    "Capture::Tiny" => 0,
    "Cwd" => 0,
    "File::Basename" => 0,
    "File::Which" => "1.21",
    "Module::Load" => 0,
    "Test::More" => 0,
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "Capture::Tiny" => 0,
  "Cwd" => 0,
  "File::Basename" => 0,
  "File::Which" => "1.21",
  "Module::Load" => 0,
  "Test::More" => 0,
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Alien::Base::ModuleBuild->new(%module_build_args);


$build->create_build_script;
